### ISSUE TITLE
Generate cataloging records for W1NLM1751 - W1NLM2225

### ISSUE THREAD

**Issue by TBRC-Travis** [buda-base/archive-ops#108]
The next round of NLM web images are being synced so we'll need to create the next batch of NLM catalog records accordingly, up through W1NLM2225.

I've created a "batch6" tab in the Normalized Accessions spreadsheet.  (will email link)

New batch will be from W1NLM1751 - W1NLM2225.  

I've already had Karma review the titles so we should be all set to run the import on this.

**Comment by xristy** [comment]
@karmagongde and @TBRC-Travis In reviewing the batch 6 data, I see that column G, `Print Type`, has two distinct values, `dbu can` and `Block print` - which mean the same thing and hence are almost certainly incorrect.

I went back and looked at batches 3, 4, and 5 and see that the values that appear are, `Block print` and `dbu med manuscript`.

Batch 6 column G needs to be reviewed before proceeding further

**Comment by TBRC-Travis** [comment]
@karmagongde can you take a look this?  if you need to make edits please modify the values in "BDRC-ACIP-Catalog-SingleVolumeWorks" not in the batch6 worksheet.  thank you!

**Comment by TBRC-Travis** [comment]
@xristy seems like batch6 worksheet is good as is.  please go ahead.

**Comment by xristy** [comment]
Two things:

1) @karmagongde when do we use _printType_ `blockprint` versus `dbuCan`?

2) Unfortunately, I just found an error in the NLM importer. The `/db/modules/gwt/works-nlm-import.xqm` checked only for exact values of `Block Print` and `manuscript` which were the values used in the early part of batch 1.

    Since later in batch 1 and on through 2-5, the values have been `Block Print` and `dbu med manuscript` which I did not realize until now.

    The importer incorrectly gave a _printType_ of `dbuCan` instead of `dbuMed` for the value `manuscript` in column G, and for column G values of `dbu med manuscript` no _printType_ was added to the pubInfo records.

    Until now there have been no records in the batches 1 - 5 that have a value of `dbu can` so until now all NLM records that have _printType_ `dbuCan` should be changed to `dbuMed` and the NLM records that have no _printType_ should have _printType_ `dbuMed` added.

    These changes need to be made prior to importing batch 6 since it will help to be able to test each NLM pubinfo record for the presence of `dbuCan` and change it to `dbuMed`.

    I can work on writing a script to correct for this situation and then move on to importing batch 6.

    I quite disappointed that neither I nor anyone else has noticed these issue before now.

**Comment by karmagongde** [comment]
Please make Column G, Print Type, dbu can, and Block print. All the manuscripts texts in NLM are dbu can script. we did not see any dbu med manuscripts. During the full cataloging of these work numbers, the librarian checks all the catalog information and modify accordingly.

**Comment by xristy** [comment]
@karmagongde when do we use printType blockprint versus dbuCan?

**Comment by karmagongde** [comment]
@xristy The _printType_ blockprint is using for all the woodblocks printed texts.
The _printType_ dbu can is using for all the handwritten dbu can manuscript texts.
The _printType_ dbu med is using for all the dbu med and cursive manuscript texts. So far the librarians did not found and dbu med texts in the NLM collection.

**Comment by xristy** [comment]
@karmagongde so you are you saying that when column G in a batch sheet contains `dbu med manuscript` or just `manuscript` that it is actually _printType_ `dbuCan` - I'll add a note to that effect in the script.

And so the records that were imported without a _printType_ (which were ones for which the importer incorrectly ignored `dbu med manuscript`) should have _printType_ `dbuCan` added to the pubinfo?

**Comment by karmagongde** [comment]
Exactly, the column G in a batch spreadsheet dbu me manuscript are printType dbuCan. 

The earlier batches 1-4 are already added _printType dbuCan_ by librarians when we did the full cataloging. Now we are full cataloging batch 5 any many work numbers are checked out on DLMS. So we will add _printType dbuCan_ for all the missing one.

Form the batch no. 6 the import of _printType dbuCan_ will be great. Thank you.

**Comment by xristy** [comment]
Ok so I have corrected the import so that it properly detects variants of "dbu can", "dbu med", "manuscript" and assigns _printType_ `dbuCan` in all of those cases:
```
    let $ptyp := 
        if (matches($prnTypeStr, "block.*print", "i")) then
            "blockprint"
        else if (matches($prnTypeStr, "dbu.*med|manuscript", "i")) then
            "dbuCan"
        else if (matches($prnTypeStr, "dbu.*can", "i")) then
            "dbuCan"
        else
            ()
```

If in the future some actual "dbu med" are discovered please open an issue to alert that there needs to be attention to treating Column G differently on import.

Further, I will not do anything with the earlier records that have/had no _printType_.

**Comment by karmagongde** [comment]
Sure, I will inform and alert if there is any actual _dbu med printType_.
 Thank you.

**Comment by xristy** [comment]
@karmagongde In the following entries Column B, "No. of Volumes", is empty which is not currently allowed in the import script:
```
W1NLM1764
W1NLM1766
W1NLM1769
W1NLM1774
W1NLM1776
W1NLM1779
```
What should I do?

It would be good if another pair of eyes took a look at: [test run](https://www.tbrc.org/docs/nlm/summary-BDRC-NLM-records-batch_06-20200619-2020-06-19T19:36:39.258Z.txt) to see if there are other problems.

**Comment by karmagongde** [comment]
@xristy I just added 1 on Column D "Vol." in the main spreadsheet for those missing and checked back and forth on the whole work number of this set. Now looks complete.

**Comment by xristy** [comment]
I made the updates to be 1 volume for each of 6 works listed above and have run the import. [View the import log](https://www.tbrc.org/docs/nlm/summary-BDRC-NLM-records-batch_06-20200619-2020-06-19T21:23:53.36Z.txt).

**Comment by karmagongde** [comment]
It is perfect, now it imported the _printType_ "dbuCan" 
Thank you Chris.

**Comment by TBRC-Travis** [comment]
I see the records on the website now.  Thanks Chris.
